### PR TITLE
Don't always focus pane content on Tapped, if the pane is already focused

### LINKED ISSUE #17062: [1.21] Clicking on the right side of a combobox doesn't cause it to open / dropdown
### Windows Terminal version

1.21.1011.0

### Windows build number

10.0.19045.3448

### Other Software

_No response_

### Steps to reproduce

look the video
https://github.com/microsoft/terminal/assets/11919060/f5c52bf2-d51c-482c-9dfc-557d919ec554

ps. for future is good idea post issue like BUG for CANARY builds WT?

### Expected Behavior

good work wit combobox 
latest good canary buid:  terminal-1.21.931.0

### Actual Behavior

terminal-1.21.991.0
terminal-1.21.1011.0
is bad for work with combobox

### PR DESCRIPTION
**Pull request by zadjii-msft**
You'll never believe this. Clicking on the dropdown button on a ComboBox doesn't set `e.Tapped = true`. It bubbles up, and lands in our `Pane`'s `Border`'s tapped handler. And in there, we yeet focus to the first content. We end up stealing focus from the combobox, and then the combobox doesn't actually open its dropdown. 

So yea we can just fix that. Easy enough. 

Closes #17062

### CODE REVIEW

File: src/cascadia/TerminalSettingsEditor/GlobalAppearance.xaml [hunk 1]
```diff
@@ -31,7 +31,8 @@
         <local:SettingContainer x:Uid="Globals_Language"
                                 Visibility="{x:Bind ViewModel.LanguageSelectorAvailable}">
             <ComboBox ItemsSource="{x:Bind ViewModel.LanguageList}"
-                      SelectedItem="{x:Bind ViewModel.CurrentLanguage, Mode=TwoWay}">
+                      SelectedItem="{x:Bind ViewModel.CurrentLanguage, Mode=TwoWay}"
+                      Style="{StaticResource ComboBoxSettingStyle}">
```
Comment (zadjii-msft) [inline src/cascadia/TerminalSettingsEditor/GlobalAppearance.xaml:35]: This is a drive-by that I discovered while investigating.

File: scratch/ScratchIslandApp/SampleApp/MySettings.h [hunk 2]
```diff
@@ -11,7 +11,7 @@ Licensed under the MIT license.
 #include <conattrs.hpp>
 #include "MySettings.g.h"
 
-using IFontFeatureMap = winrt::Windows::Foundation::Collections::IMap<winrt::hstring, uint32_t>;
+using IFontFeatureMap = winrt::Windows::Foundation::Collections::IMap<winrt::hstring, float>;
```
Comment (zadjii-msft) [inline scratch/ScratchIslandApp/SampleApp/MySettings.h:14]: Another drive-by to fix the scratch app build

File: src/cascadia/TerminalApp/Pane.cpp [hunk 3]
```diff
@@ -47,14 +47,8 @@ Pane::Pane(const IPaneContent& content, const bool lastFocused) :
...
-    _borderFirst.Tapped([this](auto&, auto& e) {
-        _FocusFirstChild();
-        e.Handled(true);
-    });
-    _borderSecond.Tapped([this](auto&, auto& e) {
-        _FocusFirstChild();
-        e.Handled(true);
-    });
+    _borderFirst.Tapped([this](auto& s, auto& e) { _borderTappedHandler(s, e); });
+    _borderSecond.Tapped([this](auto& s, auto& e) { _borderTappedHandler(s, e); });
```
Comment (lhecker) [inline src/cascadia/TerminalApp/Pane.cpp:51]:
```suggestion
    _borderFirst.Tapped({ this, _borderTappedHandler });
    _borderSecond.Tapped({ this, _borderTappedHandler });
```
The construction without a lambda is more efficient, technically. But it's also a bit easier to read IMO.

File: src/cascadia/TerminalApp/Pane.cpp [hunk 4]
```diff
@@ -88,14 +82,8 @@ Pane::Pane(std::shared_ptr<Pane> first,
...
```
Comment (lhecker) [inline src/cascadia/TerminalApp/Pane.cpp:86]:
```suggestion
    _borderFirst.Tapped({ this, _borderTappedHandler });
    _borderSecond.Tapped({ this, _borderTappedHandler });
```

**Review by lhecker — APPROVED** [review]

File: src/cascadia/TerminalApp/Pane.cpp [hunk 5]
```diff
@@ -1237,6 +1225,14 @@ void Pane::UpdateVisuals()
 // - <none>
 void Pane::_Focus()
 {
+    // Don't focus our content if we're already focused. This prevents a bug
+    // where tapping on the arrow in a ComboBox will land in our Tapped handler,
+    // and if we steal focus from the ComboBox, it won't open. See GH#17062
+    if (WasLastFocused())
```
Comment (DHowett) [inline src/cascadia/TerminalApp/Pane.cpp:1231]: Is there a way for us to check if focus **is currently inside** our pane? Rather than us trying to remember it.

**Review by DHowett — APPROVED** [review]

**Review by DHowett — APPROVED** [review]

**Review by carlos-zamora — APPROVED** [review]

File: src/cascadia/TerminalApp/Pane.cpp [hunk 6]
```diff
@@ -47,14 +47,8 @@ Pane::Pane(const IPaneContent& content, const bool lastFocused) :
...
-    _borderFirst.Tapped([this](auto&, auto& e) {
-        _FocusFirstChild();
-        e.Handled(true);
-    });
-    _borderSecond.Tapped([this](auto&, auto& e) {
-        _FocusFirstChild();
-        e.Handled(true);
-    });
+    _borderFirst.Tapped({ this, _borderTappedHandler });
+    _borderSecond.Tapped({ this, _borderTappedHandler });
```
Comment (DHowett) [inline src/cascadia/TerminalApp/Pane.cpp:51]:
```suggestion
    _borderFirst.Tapped({ this, &Pane::_borderTappedHandler });
    _borderSecond.Tapped({ this, &Pane::_borderTappedHandler });
```

File: src/cascadia/TerminalApp/Pane.cpp [hunk 7]
```diff
@@ -88,14 +82,8 @@ Pane::Pane(std::shared_ptr<Pane> first,
...
```
Comment (DHowett) [inline src/cascadia/TerminalApp/Pane.cpp:86]:
```suggestion
    _borderFirst.Tapped({ this, &Pane::_borderTappedHandler });
    _borderSecond.Tapped({ this, &Pane::_borderTappedHandler });
```

**Review by lhecker — APPROVED** [review]
